mocha-casperjs.js

```diff
@@ -1,15 +1,37 @@
 module.exports = function (Mocha, casper, utils) {
   var currentDone,
       currentTest,
+      currentError,
       f = utils.format,
 
   failTest = function(error) {
     casper.unwait()
     clearInterval(casper.checker)
-    casper.checker = null
-    if (currentDone && (!currentTest || !currentTest.state)) {
-      currentDone(error)
+
+    // the first error takes priority
+    if ( ! currentError ) {
+        currentError = error;
+    }
+    if ( currentTest.errors === undefined ) {
```

Comment:
this is the mocha test object, so not sure about such a generic `errors` property. perhaps `stepErrors` ?